finality-aleph/src/validator_network/service.rs

```diff
@@ -150,76 +154,122 @@ impl<D: Data, A: Data + Debug, ND: Dialer<A>, NL: Listener> Service<D, A, ND, NL
             });
     }
 
+    fn peer_addresses(&self, peer_id: &AuthorityId) -> Option<Vec<A>> {
+        match self.legacy_connected.contains(peer_id) {
```

Comment:
shouldn't we prefer connection with new protocol?

Reply:
We do, but the logic for that is in other parts of the code – if `legacy_connected` contains a `peer_id` this means we are now convinced that we have to use the legacy protocol to communicate with this peer until further notice.